enterprise/admin/views.py

```diff
@@ -291,132 +292,396 @@ def _handle_bulk_upload(cls, enterprise_customer, manage_learners_form, request,
 
         return all_processable_emails
 
-    # TODO: this method is HUGE (> 100 lines) and complicated. Refactor it.
-    # Potential improvements:
-    # 1. Restructure flow so that notifications are sent after users and pending users are enrolled
-    #    FIXME: `send_email_notification_message` for pending users are executed OUTSIDE of ConditionalEmailConnection
-    #    context manager.
-    # 2. Split enrolling existing users and non-existing (pending) users into methods.
-    # 3. Extract notifications into dedicated method
     @classmethod
-    def _enroll_users(cls, enterprise_customer, emails, course_id, mode, request, notify=True):
+    def enroll_user(cls, enterprise_customer, user, course_mode, *course_ids):
         """
-        Enroll the users with the given email addresses to the course specified by course_id.
+        Enroll a single user in any number of courses using a particular course mode.
 
         Args:
-            cls (type): The EnterpriseCustomerManageLearnersView class itself
-            enterprise_customer: The instance of EnterpriseCustomer whose attached users we're enrolling
-            emails: An iterable of strings containing email addresses to enroll in a course
-            course_id: The ID of the course in which we want to enroll
-            mode: The enrollment mode the users will be enrolled in the course with
-            request: The HTTP request the enrollment is being created by
-            notify: Whether to notify (by email) the users that have been enrolled
+            enterprise_customer: The EnterpriseCustomer which is sponsoring the enrollment
+            user: The user who needs to be enrolled in the course
+            course_mode: The mode with which the enrollment should be created
+            *course_ids: An iterable containing any number of course IDs to eventually enroll the user in.
+
+        Returns:
+            Boolean: Whether or not enrollment succeeded for all courses specified
         """
-        enrolled = []
-        non_existing = []
-        failed = []
+        enterprise_customer_user, __ = EnterpriseCustomerUser.objects.get_or_create(
+            enterprise_customer=enterprise_customer,
+            user_id=user.id
+        )
         enrollment_client = EnrollmentApiClient()
-        course_details = CourseCatalogApiClient(request.user).get_course_run(course_id)
+        succeeded = True
+        for course_id in course_ids:
+            try:
+                enrollment_client.enroll_user_in_course(user.username, course_id, course_mode)
+            except HttpClientError as exc:
+                succeeded = False
+                error_message = json.loads(exc.content.decode()).get('message', 'No error message provided')
```

Comment:
@haikuginger `json.loads(exc.content.decode())` might be prone to crashes in case HttpClientError response is not json (i.e. for 500 or 404). I would suggest wrapping it into another try.